packages/svelte/tests/signals/test.ts

```diff
@@ -402,6 +402,43 @@ describe('signals', () => {
 		};
 	});
 
+	test('schedules rerun when writing to signal before reading it from derived', (runes) => {
+		if (!runes) return () => {};
+		let log: any[] = [];
+
+		const value = state(1);
+		const double = derived(() => $.get(value) * 2);
+
+		user_effect(() => {
+			set(value, 10);
+			log.push($.get(double));
+			set(value, 10);
```

Comment:
Why is there another `set` afterwards? Isn't this masking what the test is actually trying to test?
```suggestion
```